src/SIMD.jl

```diff
@@ -121,7 +121,8 @@ end
 (::Type{Vec}){N,T<:ScalarTypes}(xs::NTuple{N,T}) = Vec{N,T}(xs)
 
 # Convert between vectors
-@inline Base.convert{N,T}(::Type{Vec{N,T}}, v::Vec{N}) = Vec{N,T}(Tuple(v))
+@inline Base.convert{N,T1, T2}(::Type{Vec{N,T1}}, v::Vec{N, T2}) = Vec{N,T1}(Tuple(v))
```

Comment:
I don't think it's necessary to introduce the type `T2` here; the original function could remain as is.

Did you look at the generated code -- is it actually smaller / faster?